src/core/PlaneWaveBasis.jl

```diff
@@ -100,43 +105,30 @@ end
 build_kpoints(basis::PlaneWaveBasis, kcoords) =
     build_kpoints(basis.model, basis.fft_size, kcoords, basis.Ecut)
 
-"""
-Plan a FFT of type `T` and size `fft_size`, spending some time on finding an optimal algorithm
-"""
-function build_fft_plans(T, fft_size)
-    tmp = Array{Complex{T}}(undef, fft_size...)
-
-    flags = FFTW.MEASURE
-    if T == Float32
-        flags |= FFTW.UNALIGNED
-        # TODO For Float32 there are issues with aligned FFTW plans.
-        #      Using unaligned FFTW plans is discouraged, but we do it anyways
-        #      here as a quick fix. We should reconsider this in favour of using
-        #      a parallel wisdom anyways in the future.
-    end
-    ipFFT = plan_fft!(tmp, flags=flags)
-    opFFT = plan_fft(tmp, flags=flags)
-    ipFFT, opFFT
-end
-
 """
 TODO docme
 
 fft_size is now Fourier grid size
 kcoords is vector of Vec3
 """
 function PlaneWaveBasis(model::Model{T}, Ecut::Number,
-                        kcoords::AbstractVector, ksymops=nothing, kweights=nothing; fft_size=nothing) where {T <: Real}
-    ## TODO this constructor is too low-level. Write a hierarchy of
-    ## constructors starting at the high level
-    ## `PlaneWaveBasis(model, Ecut, kgrid)`
+                        kcoords::AbstractVector, ksymops=nothing, kweights=nothing;
+                        fft_size=nothing) where {T <: Real}
+    # TODO this constructor is too low-level. Write a hierarchy of
+    # constructors starting at the high level
+    # `PlaneWaveBasis(model, Ecut, kgrid)`
 
     @assert Ecut > 0
     if fft_size === nothing
         fft_size = determine_grid_size(model, Ecut)
     end
-
     fft_size = Tuple{Int, Int, Int}(fft_size)
+
+    # TODO generic FFT is kind of broken for some fft sizes
+    if !(T in [Float32, Float64]) && !all(is_fft_size_ok_for_generic.(fft_size))
+        fft_size = next_working_fft_size_for_generic.(fft_size)
+        @info "Changing fft size to $fft_size (smallest working size for generic FFTs)"
+    end
```

Comment:
Move that to `determine_grid_size`

Reply:
I thought about that, but I disagree. If you select the wrong size here it's not a matter of making a numerical error it's a matter of actually getting a completely wrong result: It's an implementation detail specific to the generic FFT case and will (hopefully) disappear. It needs to be done at this place, however, since in the `build_fft_plans` I cannot actualy *change* `fft_size`, only read it.

Reply:
OK that makes sense. These kinds of ugly temporary workarounds are unfortunately unavoidable, but fine as long as they are explicitly marked as such and isolated as much as possible from "real" code.